projects/public-search/src/app/api/holdings-api.service.ts

```diff
@@ -47,4 +54,30 @@ export class HoldingsApiService extends BaseApi {
     .getRecords('holdings', query, page, itemsPerPage, undefined, { view: viewcode }, BaseApi.reroJsonheaders)
     .pipe(map((response: Record) => response.hits));
   }
+
+  /**
+   * Holding Can request
+   * @param holdingPid - Holding pid
+   * @param libraryPid - Library pid
+   * @param patronBarcode - Patron Barcode
+   * @return Observable - Check holding can be requested
+   */
+  canRequest(holdingPid: string, libraryPid: string, patronBarcode: string): Observable<HoldingCanRequest> {
+    const url = `/api/holding/${holdingPid}/can_request`;
+    const params = new HttpParams()
+      .set('library_pid', libraryPid)
+      .set('patron_barcode', patronBarcode);
+    return this._httpClient.get<HoldingCanRequest>(url, { params });
+  }
+
+  /**
+   * Holding patron request
+   * @param data - Holding pid, pickup location pid, description
+   * @return Observable - Create provitional item
+   */
+  request(data: { holding_pid: string, pickup_location_pid: string, description: string }): Observable<HoldingPatronRequest> {
+    const url = '/api/holding/patron_request';
+    return this._httpClient.post<HoldingPatronRequest>(url, data);
+  }
```

Comment:
Return docstring tells about "provisional item" but Observable return interface `HoldingPatronRequest`.
typo in "provitional" --> "provisional"